dip-0003/network-info.md

```diff
@@ -0,0 +1,258 @@
+# Appendix C: Network Information
+
+## Legacy Format
+
+### Masternode Information
+
+The fields below are represented as `networkInfo` and are applicable to all masternodes types as defined in
+[Appendix B](masternode-types.md)
+
+| Field     | Type    | Size | Description                                                                |
+| --------- | ------- | ---- | -------------------------------------------------------------------------- |
+| ipAddress | byte[]  | 16   | IPv6 address in network byte order. Only IPv4 mapped addresses are allowed |
+| port      | uint_16 | 2    | Port (network byte order)                                                  |
+
+#### <a name="mninfo_rules">Validation Rules</a>
+
+* `ipAddress` MUST be a valid IPv4 address that is routable on the global internet
+* `ipAddress` MUST NOT be already used in the registered masternodes set
+* `port` MUST be within the valid port range [1, 65535]
+* On mainnet, `port` MUST be set to the following value
+
+  | Field  | Value  |
+  | -------| ------ |
+  | port   | 9999   |
+
+### Platform Information
+
+The fields below are represented as `platformInfo` and are only applicable for EvoNodes as defined in
+[Appendix B](masternode-types.md)
+
+| Field            | Type    | Size   | Description                                                         |
+| ---------------- | ------- | ------ | ------------------------------------------------------------------- |
+| platformP2PPort  | uint_16 | 0 or 2 | TCP port of Dash Platform peer-to-peer communication between nodes. |
+| platformHTTPPort | uint_16 | 0 or 2 | TCP port of Platform HTTP/API interface.                            |
+
+#### <a name="plinfo_rules">Validation Rules</a>
+
+* `platformP2PPort` and `platformHTTPPort` MUST be within the valid port range [1, 65535]
+* `platformP2PPort`, `platformHTTPPort`, and [`port`](#masternode-information) from `networkInfo` MUST be distinct
+* On mainnet, `platformP2PPort` and `platformHTTPPort` MUST be set to the following values
+
+  | Field              | Value  |
+  | ------------------ | ------ |
+  | platformP2PPort    | 26656  |
+  | platformHTTPPort   | 443    |
+
+## Extended Format
+
+The fields below are represented as `networkInfo` and subsume the responsibility of `platformInfo`.
+Applicability of fields based on masternode type will be defined by the field individually.
+
+| Field     | Type    | Size     | Description                                                                                    |
+| --------- | ------- | -------- | ---------------------------------------------------------------------------------------------- |
+| p_count   | uint_8  | 1        | Number of purposes for which addresses are defined                                             |
```

Comment:
maybe better to use "compact length" for this type? It takes 1 byte for small numbers, but is not limited by 255.

Same for `purpose` and `count` below

Reply:
A purpose represents a set of up to 32 entries. Currently we have three, `CORE_P2P`, `PLATFORM_HTTPS` and `PLATFORM_P2P`. If we need more than 255 _purposes_ (i.e. storing up to 8,160 unique addresses per masternode), a revisiting of the current implementation seems to be more in order.

Allowing 32 entries may also be too much and I'm looking for feedback to gauge if 32 is adequate, insufficient or excessive.